conans/client/conf/config_installer.py

```diff
@@ -111,32 +109,76 @@ def _process_download(item, client_cache, output, tmp_folder, verify_ssl, reques
-                item = client_cache.conan_config.get_item("general.config_install")
-            except ConanException:
-                raise ConanException("Called config install without arguments and "
-                                     "'general.config_install' not defined in conan.conf")
-
-        if item.endswith(".git") or config_type == "git":
-            _process_git_repo(item, client_cache, output, tmp_folder, verify_ssl, args)
-        elif os.path.isdir(item):
-            _process_folder(item, client_cache, output)
-        elif os.path.isfile(item):
-            _process_zip_file(item, client_cache, output, tmp_folder)
-        elif item.startswith("http"):
-            _process_download(item, client_cache, output, tmp_folder, verify_ssl,
+
+        if config_type == "git":
+            _process_git_repo(path_or_url, client_cache, output, tmp_folder, verify_ssl, args)
+        elif config_type == "dir":
+            args = None
+            _process_folder(path_or_url, client_cache, output)
+        elif config_type == "file":
+            args = None
+            _process_zip_file(path_or_url, client_cache, output, tmp_folder)
+        elif config_type == "url":
+            args = None
+            _process_download(path_or_url, client_cache, output, tmp_folder, verify_ssl,
                               requester=requester)
         else:
-            raise ConanException("I don't know how to process %s" % item)
+            raise ConanException("Unable to process config install: %s" % path_or_url)
     finally:
-        if item:
-            client_cache.conan_config.set_item("general.config_install", item)
+        if config_type is not None and path_or_url is not None:
+            value = "%s:[%s, %s]" % (config_type, path_or_url, args)
+            client_cache.conan_config.set_item("general.config_install", value)
         rmdir(tmp_folder)
+
+
+def _process_config_install_item(item):
+    """
+    Processes a config_install item and outputs a tuple with the configuration type, the path/url
+    and additional args
+
+    :param item: config_install item value from conan.conf with a simple path/url or a string
+    following this pattern: "<config_type>:[<path_or_url>, <args>]"
+    :return: configuration source type (git, url, dir or file), path to file/dir or git/http url and
+    additional arguments
+    """
+    config_type, path_or_url, args = None, None, None
+    if not item.startswith(("git:", "dir:", "url:", "file:")):
+        path_or_url = item
+        if path_or_url.endswith(".git"):
+            config_type = "git"
+        elif os.path.isdir(path_or_url):
+            config_type = "dir"
+        elif os.path.isfile(path_or_url):
+            config_type = "file"
+        elif path_or_url.startswith("http"):
+            config_type = "url"
+        else:
+            raise ConanException("Unable to process config install: %s" % path_or_url)
+    else:
+        config_type, rest = item.split(":", 1)
```

Comment:
I think this will break for existing data in ``conan.conf`` that it will only contain 1 item. Please check and add tests if necessary

Reply:
This PR takes into account the possibility of having still one item in the conan.conf. Have alook again at https://github.com/conan-io/conan/pull/4083/files#diff-32564897949777992206da1e43543b08R168

Reply:
Ok, you are right, I was misreading the code.